src/components/TodosFilter/TodosFilter.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useContext } from 'react';
+import cn from 'classnames';
+import { TodosContext } from '../../store/TodoProvider';
+import { FilterType } from '../../types/Todo';
+
+type Props = {
+  onChange?: (filterBy: FilterType) => void,
+};
+
+export const TodosFilter: React.FC<Props> = ({ onChange = () => {} }) => {
+  const { state } = useContext(TodosContext);
+
+  return (
+    <nav className="filter" data-cy="Filter">
+      <a
+        href="#/"
+        className={cn('filter__link', {
+          selected: state.filterBy === FilterType.ALL,
+        })}
+        onClick={() => onChange(FilterType.ALL)}
```

Comment:
suggest to make one function factory for all filter buttons and just call in jsx with proper FilterType

```
const handleFilter = (filter: FilterType) => () => onChange(filter);
```
```suggestion
        onClick={handleFilter(FilterType.ALL)}
```